### DISCUSSION TITLE
Automatic recreation of nodes. Self healing abilities

### DISCUSSION THREAD

**Discussion by HendrikLevering** [mysticaltech/terraform-hcloud-kube-hetzner#966] [Q&A]
I simulated a "node failure" by deleting 1 of the 3 control plan nodes in the Hetzner web ui. I would expect that the cluster would notice this and recreate the 3rd node automatically. However, the cluster keeps running with 2 nodes. I have to run terraform apply manually to recreate the 3rd node. 
Is this expected or should the heal itself automatically? If so: Are there any configurations I have to set to achieve this behaviour?

**Comment by dhofstetter** [comment]
@HendrikLevering If you delete a server, you delete the server. The self healing capability is not by buying you new servers ;) Instead whenever a node is down, the workload scheduled there is shifted to another node. That's it that's all. Thats the best you can get  :)

**Reply by WebSpider** [reply]
The automatic re-creation of nodes is possible, when you have cluster-autoscaler enabled, in which case the cluster-autoscaler will automatically spawn a new *worker* node for you to run workloads on.

cluster-autoscaler doesn't touch the control-plane/master nodes, that's a bit too advanced at the moment.
